Storage/REST-API.md

```diff
@@ -0,0 +1,185 @@
+   
+   However, if the type of `key` is string, the JavaScript-friendly format of this dictionary becomes an object: `{"key1":value1,"key2":value2,...}`.
+
+   `TODO(dkorolev): Check if some other F#-related issues should be added into this section.`
+
+### Views
+
+**View** is a representation of a record, which contains some non-empty subset of the record’s fields.
+The API supports different views via the `&format=...` URL parameter.
+
+By default, records returned by the API contain all fields.
+
+Two implicitly defined views (both default to "all fields") are `full` and `preview`.
+
+The `full` view is the default, so that `&format=full` is equivalent to not specifying the `&format` URL parameter.
+
+The `&format=preview` is used when the response schema for certain endpoint contains not one record, but a collection of them. In other words, the representation of each record of `"data":[...]` for the endpoints where `"data"` is an array is equivalent to the representation of those records with `&format=preview`.
+
+The views are defined by the user, and the default representation of any view is "all fields". In particular, unless the user defines the `"preview"` view, it will be identical to the `"full"` one.
+
+### Methods
+
+Current Storage API interprets HTTP verbs in the following way:
+
+* **GET**: Get the resource.
+
+  Strongly typed with respect to returning an individual record or a collection of records. For example, `/users` can be the way to GET the list of users (`"data":[...]`), and `/user/1` can be the way to GET an individual user with the ID of `1`.
+   
+* **PATCH**: Update some fields of the resource.
+
+  `PATCH` should be called on the URL that has been (or could have been) returned as the top-level `"url":"..."` field of the payload of `GET`.
+
+  Generally, the user is expected to retrieve the value of the object under the key `"data"` of the top-level payload of a `GET` request, alter this object, and send its modified version back to the server.
+
+  `PATCH` respects the format of the `"data"` controlled by the `&format=` URL parameter. Sending a `PATCH` request to the URL with certain `&format=X` can only modify the fields returned by the `GET` request of this `&format=X`.
+
+  For `PATCH` requests, the key of the record being updated is taken from the URL. The user is expected to either not send in the key as part of the payload, or ensure the key contained in the payload matches the key contained in the URL.
+  
+  `PATCH` will fail if attempted to modify a non-existing resource, resulting in `404 (Not Found)` error code.
+  
+  Also, `PATCH` performs conflict resolution. The URL being `PATCH`-ed must contain a revision / timestamp of the resource. The exact semantics of the revision / timestamp is not specified by this RFC, but the top-level `"url"` field returned by the `GET` request is guaranteed to have it set.
```

Comment:
No reason except for the URL is always there, and asking the user to send the mutating request onto the URL returned in the `.url` field of the GET-ted JSON looks straightforward.

The solution involving headers looks heavier to my taste (for instance, I'd have to look it up how to send a header from F# or Ruby).